app/Models/User.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace App\Models;
+
+use Illuminate\Database\Eloquent\Factories\HasFactory;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use Illuminate\Database\Eloquent\Concerns\HasUuids;
+use Illuminate\Support\Str;
+use Illuminate\Database\Eloquent\SoftDeletes;
+use Illuminate\Foundation\Auth\User as Authenticatable;
+use Illuminate\Notifications\Notifiable;
+use Tymon\JWTAuth\Contracts\JWTSubject;
+
+class User extends Authenticatable implements JWTSubject
+{
+    use HasFactory, HasUuids;
+    use Notifiable;
+    use SoftDeletes;
+
+    const EMAIL = 'email';
+    const FULL_NAME = 'full_name';
+    const PASSWORD = 'password';
+    const ROLE = 'role';
+
+    protected $fillable = [
+        self::EMAIL,
+        self::FULL_NAME,
+        self::PASSWORD,
+        self::ROLE,
+    ];
+    
+    protected $hidden = [
+        'password'
+    ];
+
+    protected $dates = ['deleted_at'];
```

Comment:
The dates property is deprecated in Laravel 8 and above. Use $casts instead with deleted_at cast to datetime.
```suggestion
    protected $casts = [
        'deleted_at' => 'datetime',
    ];
```